courses-rest/src/main/java/no/sikt/nva/fs/CoursesProvider.java

```diff
@@ -0,0 +1,110 @@
+package no.sikt.nva.fs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.text.Collator;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import no.sikt.nva.fs.client.FsClient;
+import no.sikt.nva.fs.client.FsClientException;
+import no.sikt.nva.fs.client.FsCourse;
+import no.sikt.nva.fs.client.FsSemester;
+import no.sikt.nva.fs.client.HttpUrlConnectionFsClient;
+import no.sikt.nva.fs.client.Item;
+import no.sikt.nva.fs.config.InstitutionConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CoursesProvider {
```

Comment:
Is this code copied from somewhere else?

Reply:
Not really. Looking at it, I am refactoring it for readability.